server/src/strategy.js

```diff
@@ -0,0 +1,73 @@
+const passport = require('passport');
+const { Strategy } = require('passport-discord');
+const axios = require('axios');
+
+const User = require('./database/models/User');
+const Guild = require('./database/models/Guild');
+
+passport.use(new Strategy({
+  clientID: process.env.CLIENT_ID,
+  clientSecret: process.env.CLIENT_SECRET,
+  callbackURL: process.env.CLIENT_CALLBACK_URL,
+  scope: ['identify', 'guilds'],
+}, async (accessToken, refreshToken, profile, done) => {
+  try {
+    let user = await User.findOne({ userId: profile.id });
+    if (!user) {
+      const userGuilds = profile.guilds.filter(async (g) => {
+        const guilds = await axios.get(`https://discord.com/api/v6/users/${process.env.CLIENT_ID}/guilds`, {
```

Comment:
update the url to be https://discord.com/api/v6/users/@me/guilds for both conditions

Reply:
else it returns 404 not found